packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -725,14 +725,24 @@ class RenderParagraph extends RenderBox with ContainerRenderObjectMixin<RenderBo
   @override
   bool hitTestChildren(BoxHitTestResult result, { required Offset position }) {
     final TextPosition textPosition = _textPainter.getPositionForOffset(position);
-    final Object? span = _textPainter.text!.getSpanForPosition(textPosition);
-    if (span is HitTestTarget) {
-      result.add(HitTestEntry(span));
-      return true;
+    if (_positionIsWithinText(position: position, textPosition: textPosition)) {
+      final InlineSpan? span = _textPainter.text!.getSpanForPosition(textPosition);
```

Comment:
The reason `span` was typed `Object?` was to avoid the down-casting below (i.e. `span! as HitTestTarget`).

Reply:
Makes sense, please also add a comment on this line to explain this.